languages/ocaml.scm

```diff
@@ -350,13 +350,16 @@
   "type" @prepend_space
 )
 (
-  [
-    "("
-    "["
-  ]* @do_nothing
+  (_)
   .
   (type_variable) @prepend_space
 )
+(
+  (type_variable)
+  .
+  "." @append_space
+)
+
```

Comment:
I do really understand the reason for this modification, and the one above. Could you clarify?

Reply:
(Disclaimer: I don't know the OCaml terminology, so I'll have to spell things out verbosely :sweat_smile:)

The original rule:
```scm
(
  [
    "("
    "["
  ]* @do_nothing
  .
  (type_variable) @prepend_space
)
```
..._should_ have been enough to address the spacing in the `type t: ('x -> x)`-case. However, it clearly wasn't working. As such, I changed it to this:

```scm
(
  (_)
  .
  (type_variable) @prepend_space
)
```

That worked, but it broke the spacing of type variables followed by a full stop, which is an anonymous node. i.e.:
```ocaml
(* This... *)
let my_const : 'a 'b. 'a -> 'b -> 'a = Fun.const

(* Got formatted as... *)
let my_const : 'a 'b.'a -> 'b -> 'a = Fun.const
```

I tried changing the first half of the tuple pattern to `[(_) "."]`, but that didn't work either. As such, I had to write an explicit rule for it to target type variables followed by a full stop, hence:
```scm
(
  (type_variable)
  .
  "." @append_space
)
```

There may be a better solution, but this worked.

Reply:
I think that, for the sake of the maintainability of the query file, it's better to remove from existing rules than add new rules, and I think the problem can be addressed by the former. Let me draft something

Reply:
My attempt is at #258

Reply:
That looks great :100: I'll close this PR